sympy/ntheory/digits.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function, division
+
+import math
+import re
+
+from collections import Counter
+
+
+def count_digits(n, base=10):
+    """
+    A frequency counter for the digits of an integer ``n`` given in a base
+    ``base``.
+    """
+    base_func = {2: bin, 8: oct, 10: str, 16: hex}
+
+    if base in [2, 8, 10, 16]:
+        s = re.sub(r'0[a-z]{1}', '', base_func[base](n))
+        count = Counter(s)
+        return {int(k, base=16): v for k, v in count.items()}
+
+    m = math.floor(math.log(n, base)) + 1
+    count = dict()
+    for k in range(m):
+        d = (n // base ** k) % base
+        if d in count:
+            count[d] += 1
+        else:
+            count[d] = 1
+
```

Comment:
What about?
```python
n_copy = n
while n_copy > 0:
    d = n_copy%base
    n_copy = n_copy//base
    if count.get(d, None) is None:
        count[d] = 0
    count[d] += 1
```

Reply:
There is an even simpler method

    ctr = Counter(
        ((n // base ** k) % base for k in range(m))
    )

Reply:
Which is computationally more efficient?

Reply:
I haven't tested it - I suppose you could use `timeit` to test the different versions.